crates/polars-ops/src/series/ops/is_in.rs

```diff
@@ -556,6 +560,26 @@ fn is_in_struct(ca_in: &StructChunked, other: &Series) -> PolarsResult<BooleanCh
     }
 }
 
+#[cfg(feature = "dtype-categorical")]
+fn is_in_string_categorical(
+    ca_in: &StringChunked,
+    other: &CategoricalChunked,
+) -> PolarsResult<BooleanChunked> {
+    // We need only check against the unique categories used by this array.
+    let set = if other._can_fast_unique() {
+        // Build hash set directly from categories
+        let cats = other.get_rev_map().get_categories();
+        PlHashSet::from_iter(cats.values_iter().map(|s| s.to_owned()))
+    } else {
+        // Build hash set from unique values
+        let cats = other.unique().unwrap();
+        PlHashSet::from_iter(cats.iter_str().flatten().map(|s| s.to_owned()))
```

Comment:
This copies the strings which can be expensive. 
You could slice the underlying categories which would probably be faster.

Something like
```
fn is_in_string_categorical(
    ca_in: &StringChunked,
    other: &CategoricalChunked,
) -> PolarsResult<BooleanChunked> {
    // In case of fast unique, we can directly use the categories. Otherwise we need to
    // first get the unique physicals
    let other = if other._can_fast_unique() {
        StringChunked::with_chunk("", other.get_rev_map().get_categories().clone())
    } else {
        let categories =
            StringChunked::with_chunk("", other.get_rev_map().get_categories().clone());
        let s = other.physical().unique()?.cast(&IDX_DTYPE)?;
        // SAFETY: Invariant of categorical means indices are in bound
        unsafe { categories.take_unchecked(s.idx()?) }
    };
    is_in_helper_ca(&ca_in.as_binary(), &other.as_binary())
}
```

This follows the 'regular' `is_in` flow a bit more. Your solution does make better use of the fact that there are no nulls in `cats`. @ritchie46, what do you think?

Reply:
@c-peters could we make this work on the categorical physicals? Or isn't it guaranteed that they are from the same rev-map?

If so, that would be much faster as the `hash` and `eq` operator is much cheaper for the `u32`s.

Reply:
@ritchie46 we're looking for strings in a categorical set. Converting the strings to `u32` would requiring hashing all strings anyway (right?), so we wouldn't benefit at by converting them to categorical first. I'm pretty sure the fastest way is to get a hash set of the unique values of the categoricals, and check the strings against that set.

We can save a *little* bit of time if we know the categories can be fast-uniqued in that we can just return the revmap strings and hash those. Otherwise we have to run unique on the `u32` values, then convert those to string, then hash that final set.

Reply:
Oh, right sorry for the confusion. Then I would follow @c-peters comment.